omod-2.2/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_2/ObsController2_2Test.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.controller.openmrs2_2;
+
+import org.junit.Test;
+import org.openmrs.Obs;
+import org.openmrs.api.ObsService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.web.RestTestConstants1_8;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class ObsController2_2Test extends MainResourceControllerTest {
+
+    private static final DateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS");
+
+    /**
+     * @see MainResourceControllerTest#getURI()
+     */
+    @Override
+    public String getURI() {
+        return "obs";
+    }
+
+    @Override
+    public long getAllCount() {
+        return Context.getObsService().getObservationCount(null, true);
+    }
+
+    /**
+     * @see MainResourceControllerTest#getUuid()
+     */
+    @Override
+    public String getUuid() {
+        return RestTestConstants1_8.OBS_UUID;
+    }
+
+    /**
+     * @see MainResourceControllerTest#shouldGetAll()
+     */
+    @Override
+    @Test(expected = ResourceDoesNotSupportOperationException.class)
+    public void shouldGetAll() throws Exception {
+        super.shouldGetAll();
+    }
+
+    @Test
+    public void shouldUpdateObs() throws Exception {
+        ObsService obsService = Context.getObsService();
+        final String obsUuid ="47f18998-96cc-11e0-8d6b-9b9415a91465";
+        Date updatedObsDatetime = new Date();
+        double updatedValue = 10.0 ;
+
+        executeDataSet("obsDataSet2_2.xml");
+        Obs existingObs = obsService.getObsByUuid(obsUuid);
+        assertNotNull(existingObs);
+
+        String json = "{ \"value\":\"" + updatedValue + "\", \"person\":\"" + RestTestConstants1_8.PERSON_UUID
+                + "\", \"concept\":\"SNOMED CT:2332523\",\"obsDatetime\":\""+ DATE_FORMAT.format (updatedObsDatetime) +"\",\"voided\":false}";
+
+        handle(newPostRequest(getURI() + "/" + existingObs.getUuid(), json));
+        Obs updatedObs = obsService.getRevisionObs(existingObs);
+
+        assertNotNull(updatedObs);
+        assertEquals(10.0,updatedObs.getValueNumeric(),updatedValue);
```

Comment:
What is the third parameter, above, represent?

Reply:
>assertEquals(double expected, double actual)
          Deprecated. Use assertEquals(double expected, double actual, double delta) instead

Due to https://junit.org/junit4/javadoc/4.13/org/junit/Assert.html
where double delta
due to https://www1.udel.edu/CIS/software/dist/junit4.1/javadoc/org/junit/Assert.html

>assertEquals
public static void assertEquals(double expected, double actual,double delta)

Asserts that two doubles are equal to within a positive delta. If they are not, an AssertionError is thrown. If the expected value is infinity then the delta value is ignored.NaNs are considered equal: assertEquals(Double.NaN, Double.NaN, *) passes
Parameters:
>expected - expected value
actual - the value to check against expected
delta - the maximum delta between expected and actual for which both numbers are still considered equal.

Reply:
Any what value are you passing as that third parameter? Does it make sense as a delta? ("Delta", in this case, is the value so that `abs(expected - actual) <= delta`; in other words, delta expresses the allowable difference between the two values so that they can be considered equal because sometimes, with floating point arithmetic, `1.0 != 1.000000001`).

Reply:
Ouk ......... so can I use something like `40.5` It is like am lost some how ....or may be better I use `assertEquals(expected.tostring() ,actual.tostring());`

Reply:
now a little confused @ibacher 
>using AssertEquals(expected.toString() ,actual.toString())

Will it be helpful

Reply:
@Muta-Jonathan Basically, you want:

```java
assertEquals(10.0, updateObs.getValueNumeric(), 0.001)
```

This will accept that the values are equal if `getValueNumeric()` returns a response in the range 9.999 – 10.001. This helps deal with things like:

```
jshell> 0.7 - 0.2
$1 ==> 0.49999999999999994
```

So that in your test, something like:

```java
assertEquals(0.5, 0.7 - 0.2)
```

Would fail since 0.49999999999999994 is less than 0.5, but

```java
// close enough
assertEquals(0.5, 0.7 - 0.2, 0.0001)
```

Succeeds.

Reply:
Thanks, @ibacher ...let me update my PR